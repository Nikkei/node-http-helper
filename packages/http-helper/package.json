{
    "name": "@nikkei/http-helper",
    "version": "2.4.1",
    "type": "module",
    "files": [
        "__dist/",
        "LICENSE",
        "package.json",
        "README.md"
    ],
    "typings": "./__dist/mod.d.ts",
    "exports": {
        ".": {
            "types": "./__dist/mod.d.ts",
            "default": "./__dist/mod.js"
        },
        "./auth": {
            "types": "./__dist/auth/mod.d.ts",
            "default": "./__dist/auth/mod.js"
        },
        "./cache": {
            "types": "./__dist/cache/mod.d.ts",
            "default": "./__dist/cache/mod.js"
        },
        "./csp": {
            "types": "./__dist/csp/mod.d.ts",
            "default": "./__dist/csp/mod.js"
        },
        "./sec_fetch": {
            "types": "./__dist/sec_fetch/mod.d.ts",
            "default": "./__dist/sec_fetch/mod.js"
        }
    },
    "scripts": {
        "clean": "npm run clean:dist && npm run clean:tsbuildinfo",
        "clean:dist": "rm -rf __dist/",
        "clean:tsbuildinfo": "rm -rf tsconfig.tsbuildinfo",
        "prebuild": "npm run clean",
        "build": "tsc --project ."
    },
    "repository": {
        "type": "git",
        "url": "git+https://github.com/Nikkei/node-http-helper.git"
    },
    "author": "Nikkei Inc.",
    "license": "MIT",
    "bugs": {
        "url": "https://github.com/Nikkei/node-http-helper/issues"
    },
    "homepage": "https://github.com/Nikkei/node-http-helper",
    "dependencies": {
        "option-t": "^39.0.2"
    }
}
