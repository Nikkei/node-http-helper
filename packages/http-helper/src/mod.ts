export * as Auth from './auth/mod.js';
export * as Cache from './cache/mod.js';
export * as ClearSiteData from './clear_site_data.js';
export * as CSP from './csp/mod.js';
export * as HttpHeaderName from './http_header_name.js';
export * as HttpMethod from './http_method.js';
export * as HttpStatus from './http_status_code.js';
export * as Mime from './mime.js';
export { SecFetchDest, SecFetchMode, SecFetchSite, SecFetchUser } from './sec_fetch/mod.js';
