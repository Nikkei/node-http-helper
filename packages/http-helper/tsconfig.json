{
    "extends": "./tsconfig.base.json",
    "compilerOptions": {
        "outDir": "./__obj"
    },
    "include": [
        // @prettier-ignore
        "src/**/*",
        "__tests__/**/*"
    ]
}
