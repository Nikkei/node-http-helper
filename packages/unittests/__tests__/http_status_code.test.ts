import test from 'ava';

import { HttpStatus } from '@nikkei/http-helper';

test('HttpStatus exported items', (t) => {
    t.snapshot(HttpStatus);
});
