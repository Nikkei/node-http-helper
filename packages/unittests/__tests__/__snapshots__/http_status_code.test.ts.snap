// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HttpStatus exported items 1`] = `
{
  "ACCEPTED_202": 202,
  "BAD_GATEWAY_502": 502,
  "BAD_REQUEST_400": 400,
  "CONFLICT_409": 409,
  "CONTENT_TOO_LARGE_413": 413,
  "CONTINUE_100": 100,
  "CREATED_201": 201,
  "EXPECTATION_FAILED_417": 417,
  "FORBIDDEN_403": 403,
  "FOUND_302": 302,
  "GATEWAY_TIMEOUT_504": 504,
  "GONE_410": 410,
  "HTTP_VERSION_NOT_SUPPORTED_505": 505,
  "INTERNAL_SERVER_ERROR_500": 500,
  "LENGTH_REQUIRED_411": 411,
  "METHOD_NOT_ALLOWED_405": 405,
  "MISDIRECTED_REQUEST_421": 421,
  "MOVED_PERMANENTLY_301": 301,
  "MULTIPLE_CHOICES_300": 300,
  "NETWORK_AUTHENTICATION_REQUIRED_511": 511,
  "NON_AUTHORITATIVE_INFORMATION_203": 203,
  "NOT_ACCEPTABLE_406": 406,
  "NOT_FOUND_404": 404,
  "NOT_IMPLEMENTED_501": 501,
  "NOT_MODIFIED_304": 304,
  "NO_CONTENT_204": 204,
  "OK_200": 200,
  "PARTIAL_CONTENT_206": 206,
  "PAYMENT_REQUIRED_402": 402,
  "PERMANENT_REDIRECT_308": 308,
  "PRECONDITION_FAILED_412": 412,
  "PRECONDITION_REQUIRED_428": 428,
  "PROXY_AUTHENTICATION_REQUIRED_407": 407,
  "RANGE_NOT_SATISFIABLE_416": 416,
  "REQUEST_HEADER_FIELDS_TOO_LARGE_431": 431,
  "REQUEST_TIMEOUT_408": 408,
  "RESET_CONTENT_205": 205,
  "SEE_OTHER_303": 303,
  "SERVICE_UNAVAILABLE_503": 503,
  "SWITCHING_PROTOCOLS_101": 101,
  "TEMPORARY_REDIRECT_307": 307,
  "TOO_MANY_REQUESTS_429": 429,
  "UNAUTHORIZED_401": 401,
  "UNPROCESSABLE_CONTENT_422": 422,
  "UNSUPPORTED_MEDIA_TYPE_415": 415,
  "UPGRADE_REQUIRED_426": 426,
  "URI_TOO_LONG_414": 414,
  "USE_PROXY_305": 305,
  "isClientError4xx": [Function],
  "isInformational1xx": [Function],
  "isRedirection3xx": [Function],
  "isServerError5xx": [Function],
  "isSuccessful2xx": [Function],
}
`;
